src/app/SimpleSubscriptionResumptionStorage.h

```diff
@@ -0,0 +1,143 @@
+/*
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file defines a basic implementation of SubscriptionResumptionStorage that
+ *      persists subscriptions in a flat list in TLV.
+ */
+
+#pragma once
+
+#include <app/SubscriptionResumptionStorage.h>
+#include <lib/core/TLV.h>
+#include <lib/support/DefaultStorageKeyAllocator.h>
+#include <lib/support/Pool.h>
+
+#if CHIP_CONFIG_PERSIST_SUBSCRIPTIONS
```

Comment:
How about not compiling/including this file in the build at all on this conditional.

Why are the includes above valid if `CHIP_CONFIG_PERSIST_SUBSCRIPTIONS`  is set? We effectively seem to want an empty file if CHIP_CONFIG_PERSIST_SUBSCRIPTIONS is missing. Don't do that. instead let the include be conditional not the file/cpp itself.

Reply:
Filed #24564 to track moving this conditional compilation to GN system.